Examples/ObjectiveC/SymbolLayerZOrderExample.m

```diff
@@ -0,0 +1,106 @@
+    feature1.attributes = @{@"id": @"squircle"};
+
+    MGLPointFeature *feature2 = [MGLPointFeature alloc];
+    feature2.coordinate = CLLocationCoordinate2DMake(-41.292650, 174.778768);
+    feature2.attributes = @{@"id": @"oval"};
+
+    MGLPointFeature *feature3 = [MGLPointFeature alloc];
+    feature3.coordinate = CLLocationCoordinate2DMake(-41.292650, 174.778768);
+    feature3.attributes = @{@"id": @"star"};
+
+    MGLShapeCollectionFeature *shapeCollection = [MGLShapeCollectionFeature shapeCollectionWithShapes:@[feature1, feature2, feature3]];
+    MGLShapeSource *source = [[MGLShapeSource alloc] initWithIdentifier:@"symbol-layer-z-order-example" shape:shapeCollection options:nil];
+
+    [style addSource:source];
+    self.layer = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"points-style" source:source];
+    self.layer.sourceLayerIdentifier = @"symbol-layer-z-order-example";
+    // Create a stops dictionary with keys that are possible values for 'id', paired with icon images that will represent those features.
+    NSDictionary *icons = @{@"squircle": @"squircle", @"oval": @"oval", @"star": @"star"};
+    // Use the stops dictionary to assign an icon based on the "POITYPE" for each feature.
+    self.layer.iconImageName = [NSExpression expressionWithFormat:@"FUNCTION(%@, 'valueForKeyPath:', id)", icons];
+    self.layer.iconAllowsOverlap = [NSExpression expressionForConstantValue:@(YES)];
+    self.layer.symbolZOrder = [NSExpression expressionForConstantValue:@"source"];
+    [style addLayer:self.layer];
+
+    UISegmentedControl *styleToggle =[[UISegmentedControl alloc] initWithItems:@[@"viewport-y", @"source"]];
+    styleToggle.translatesAutoresizingMaskIntoConstraints = NO;
+    styleToggle.backgroundColor = [UIColor colorWithRed:0.83 green:0.84 blue:0.95 alpha:1.0];
+    styleToggle.tintColor = [UIColor colorWithRed:0.26 green:0.39 blue:0.98 alpha:1.0];
+    styleToggle.layer.cornerRadius = 4;
+    styleToggle.clipsToBounds = YES;
+    styleToggle.selectedSegmentIndex = 1;
+    [self.view insertSubview:styleToggle aboveSubview:self.mapView];
+    [styleToggle addTarget:self action:@selector(changeStyle:) forControlEvents:UIControlEventValueChanged];
+    
+    // Configure autolayout constraints for the UISegmentedControl to align
+    // at the bottom of the map view and above the Mapbox logo and attribution
+    NSMutableArray *constraints = [NSMutableArray array];
+    
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:styleToggle attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self.mapView attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:1.0]];
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:styleToggle attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self.mapView.logoView attribute:NSLayoutAttributeTop multiplier:1 constant:-20]];
+    
+    [self.view addConstraints:constraints];
```

Comment:
I would remove all of the code that deals with setting up with toggle button into its own function and call that here instead, similar to how we do in this example: https://github.com/mapbox/ios-sdk-examples/blob/8323988d47611e9841100e72077ae46b3e0674ca/Examples/Swift/ShowHideLayerExample.swift#L18